realm/realm-library/src/objectServer/java/io/realm/internal/SyncObjectServerFacade.java

```diff
@@ -86,10 +88,22 @@ public void realmClosed(RealmConfiguration configuration) {
     public Object[] getUserAndServerUrl(RealmConfiguration config) {
         if (config instanceof SyncConfiguration) {
             SyncConfiguration syncConfig = (SyncConfiguration) config;
+            // make sure the user is still valid
+            SyncUser user = syncConfig.getUser();
+            if (!user.isValid()) {
+                if (user.getAccessToken() == null) {
+                    throw new IllegalStateException("The SyncUser is already logged out, can not use the provided configuration.");
+                } else {
+                    // user was not logged out but the `refresh_token` is not longer valid
+                    // the user will still get a stall version of Realm, that will work offline
+                    // but not sync.
+                    RealmLog.warn("SyncUser is no longer valid.");
```

Comment:
Without context, this isn't much help. Perhaps  "Opening Realm at <path>, but user <identity> is longer valid" ?